Sources/System/FilePath/FilePathTemp.swift

```diff
@@ -0,0 +1,97 @@
+/*
+ This source file is part of the Swift System open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift System project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+*/
+
+// MARK: - API
+
+/// Create a temporary path for the duration of the closure.
+///
+/// - Parameters:
+///   - basename: The base name for the temporary path.
+///   - body: The closure to execute.
+///
+/// Creates a temporary directory with a name based on the given `basename`,
+/// executes `body`, passing in the path of the created directory, then
+/// deletes the directory and all of its contents before returning.
+public func withTemporaryFilePath<R>(
```

Comment:
Make sure that any API additions go through some kind of review, ideally on the forums. At the very least, a dedicated PR that highlights it's an API addition PR as opposed to just a bug-fix PR for Windows.

For this function, would a static method on `FilePath` work instead of a global? E.g. `FilePath.withTemporaryPath { ... }`.

Reply:
Yes, it would. My only reservation is that (to me) things like `FilePath.withTemporaryPath` read like a function that _constructs and returns_ a `FilePath` from some kind of `TemporaryPath` thing. This isn't the same as e.g. `Array`'s `withUnsafeBytes`, because that's called on an instance rather than on the type itself. Maybe that's just because I'm used to ObjC. It's obvious from the arguments that it takes a closure, of course.

I only needed this for the tests, so for now I think the thing to do is to make this non-public (I'll raise a PR to fix that). I'll check through to make sure I didn't leak any other new API and do the same there.